### PR TITLE
Add build action

### PR DESCRIPTION
**Pull request by nojaf**
Ignore the first commits, I would rebase afterwards.

In 034123a9143b7f690497c793751ce375a32c46da I tried to add a way to run `checker.Compile`.

This currently doesn't really work.
`Compile` expects all the DLLs to be present. I tried building the project via an external process first, but that didn't end well (See https://github.com/dotnet/sdk/issues/9298).

And when the projects were all built before running the Generator, the FSharpProjectOption are as good as empty 😔. I'm not sure why that is, although I assume the attempt to build projects mixes up things.

In short, not sure if you really want this one 😅

### CODE REVIEW

**Comment by safesparrow** [comment]
Thanks for this idea!

>Compile expects all the DLLs to be present.

Yes, I would expect that - it behaves the same `fsc.exe` does.
During a regular commandline build MSBuild calls into `fsc.exe` as part of the `CoreCompile` target, individually for every project, once all its dependencies have been built, and passes in all the dlls.
So you'd definitely want to build all dependencies beforehand.

>I tried building the project via an external process first, but that didn't end well (See https://github.com/dotnet/sdk/issues/9298).

Hmm, what error are you seeing and what commandline were you running?

In general, it's not quite clear to me how the benchmarking should work for the compiler vs the checker.
Given that the compiler can be invoked directly from the commandline, the alternative way to run its benchmarks is to invoke `fsc.exe` directly. But maybe for now, until better setup exists, it would be handy to be able to run benchmarks of both the compiler and the checker with the same tool.

**Comment by nojaf** [comment]
Hmm, weird so that actually worked for you?
When I tried it with Fantomas, I was unable to build the project (in a separate process) before running the `checker.Compile()`. I always got exit code 1, the log said the build did succeed but no DLLs were made.

Note that I removed that code that does the build.

> How did you detect that the FSharpProjectOptions objects are invalid?

Both `SourceFiles` and `OtherOptions` were completely empty.
